pi_mqtt_gpio/modules/bme280.py

```diff
@@ -0,0 +1,195 @@
+        self.dig_H2 = 0
+        selfdig_H3 = 0
+        self.dig_H4 = 0
+        self.dig_H5 = 0
+        self.dig_H6 = 0
+        self.bus = smbus.SMBus(config["i2c_bus_num"])
+        self.address = config["chip_addr"]
+        #print("SETUP BME")
+        # Register Address
+        REG_DATA = 0xF7
+        REG_CONTROL = 0xF4
+        REG_CONFIG  = 0xF5
+        REG_CONTROL_HUM = 0xF2
+        REG_HUM_MSB = 0xFD
+        REG_HUM_LSB = 0xFE
+        # Oversample setting - page 27
+        OVERSAMPLE_TEMP = 2
+        OVERSAMPLE_PRES = 2
+        MODE = 1
+        # Oversample setting for humidity register - page 26
+        OVERSAMPLE_HUM = 2
+        self.bus.write_byte_data(self.address, REG_CONTROL_HUM, OVERSAMPLE_HUM)
+        control = OVERSAMPLE_TEMP<<5 | OVERSAMPLE_PRES<<2 | MODE
+        self.bus.write_byte_data(self.address, REG_CONTROL, control)
+        # Read blocks of calibration data from EEPROM
+        # See Page 22 data sheet
+        cal1 = self.bus.read_i2c_block_data(self.address, 0x88, 24)
+        cal2 = self.bus.read_i2c_block_data(self.address, 0xA1, 1)
+        cal3 = self.bus.read_i2c_block_data(self.address, 0xE1, 7)
+        #print("cal data:", cal1, cal2, cal3)
+        # Convert byte data to word values
+        self.dig_T1 = self.getUShort(cal1, 0)
+        self.dig_T2 = self.getShort(cal1, 2)
+        self.dig_T3 = self.getShort(cal1, 4)
+        self.dig_P1 = self.getUShort(cal1, 6)
+        self.dig_P2 = self.getShort(cal1, 8)
+        self.dig_P3 = self.getShort(cal1, 10)
+        self.dig_P4 = self.getShort(cal1, 12)
+        self.dig_P5 = self.getShort(cal1, 14)
+        self.dig_P6 = self.getShort(cal1, 16)
+        self.dig_P7 = self.getShort(cal1, 18)
+        self.dig_P8 = self.getShort(cal1, 20)
+        self.dig_P9 = self.getShort(cal1, 22)
+        self.dig_H1 = self.getUChar(cal2, 0)
+        self.dig_H2 = self.getShort(cal3, 0)
+        self.dig_H3 = self.getUChar(cal3, 2)
+
+        self.dig_H4 = self.getChar(cal3, 3)
+        self.dig_H4 = (self.dig_H4 << 24) >> 20
+        self.dig_H4 = self.dig_H4 | (self.getChar(cal3, 4) & 0x0F)
+
+        self.dig_H5 = self.getChar(cal3, 5)
+        self.dig_H5 = (self.dig_H5 << 24) >> 20
+        self.dig_H5 = self.dig_H5 | (self.getUChar(cal3, 4) >> 4 & 0x0F)
+ 
+        self.dig_H6 = self.getChar(cal3, 6)
+        #print("END SETUP")
+
+    def setup_sensor(self, config):
+        return True  # nothing to do here
+
+    def convert_to_celsius(self, value):
+        return (value / 32.0) / 8.0
+
+    def get_value(self, config):
+        """get the temperature or humidity value from the sensor"""
+        print("READING BME")
+        temperature, preassure, humidity = self.readBME280All()
+        if config["type"] == "temperature" and temperature is not None:
+            return temperature
+        if config["type"] == "humidity" and humidity is not None:
+            return humidity
+        if config["type"] == "preassure" and preassure is not None:
```

Comment:
Typo here, should be `pressure`

Reply:
Resolved with commit b7e75f044c7eb475d6ea36f648c60d8c37c73bf1